yb-voyager/src/query/queryissue/parser_issue_detector.go

```diff
@@ -206,6 +265,85 @@ func (p *ParserIssueDetector) getPLPGSQLIssues(query string) ([]QueryIssue, erro
 	}), nil
 }
 
+// FinalizeColumnMetadata processes the column metadata after all DDL statements have been parsed.
+func (p *ParserIssueDetector) FinalizeColumnMetadata() {
+	p.finalizeInheritedTableColumns()
+	p.finalizePartitionedTableColumns()
+	p.finalizeForeignKeyConstraints()
+}
+
+// finalizeInheritedTableColumns copies column metadata from parent tables to inherited child tables.
+func (p *ParserIssueDetector) finalizeInheritedTableColumns() {
+	for child, parents := range p.inheritedFrom {
+		for _, parent := range parents {
+			parentCols, ok := p.columnMetadata[parent]
+			if !ok {
+				continue // Parent metadata not found
+			}
+			if _, exists := p.columnMetadata[child]; !exists {
+				p.columnMetadata[child] = make(map[string]*ColumnMetadata)
+			}
+			for colName, colMeta := range parentCols {
+				if _, exists := p.columnMetadata[child][colName]; !exists {
+					copy := *colMeta
+					p.columnMetadata[child][colName] = &copy
+				}
+			}
+		}
+	}
```

Comment:
I think this code for resolving meta info from parent seems similar for partition and inherited, maybe you can create a single function for this